sc/shared/src/main/scala/sigmastate/lang/SigmaTemplateCompiler.scala

```diff
@@ -0,0 +1,56 @@
+package sigmastate.lang
+
+import fastparse.Parsed
+import org.ergoplatform.sdk.ContractTemplate
+import sigmastate.Values.SValue
+import sigmastate.lang.syntax.ParserException
+import org.ergoplatform.sdk.NetworkType
+import sigmastate.eval.CompiletimeIRContext
+import org.ergoplatform.sdk.Parameter
+
+/** Compiler which compiles Ergo contract templates into a [[ContractTemplate]]. */
+class SigmaTemplateCompiler(networkPrefix: Byte) {
+  val sigmaCompiler = new SigmaCompiler(networkPrefix)
+
+  /**
+   * Compiles the provided contract source code into a [[ContractTemplate]].
+   * @param source The ErgoScript contract source code.
+   * @return The parsed contract template.
+   */
+  def compile(source: String): ContractTemplate = {
+    ContractParser.parse(source) match {
+      case Parsed.Success(template, index) =>
+        val body = source.slice(index, source.length)
```

Comment:
This looks like a hack. I suggest to make the `body` to be part of ParsedContractTemplate and all the data parsed at once. Then you can create an overloaded `compile` method and pass the parsed body directly to finish compilation.

Reply:
I _think_ I've done what you're asking for here